### PR TITLE
5 datetime functions

### PR DESCRIPTION
**Pull request by izulin**
### Context

Functions: INTERVAL, NETWORKDAYS, NETWORKDAYS.INTL, WORKDAY, WORKDAY.INTL.

### Types of changes

- [ ] Bug fix (non-breaking change which fixes an issue)
- [x] New feature or improvement (non-breaking change which adds functionality)
- [ ] Breaking change (fix or feature that would cause existing functionality to not work as expected)
- [ ] Additional language file or change to the existing one (translations)

### Related issue(s):
1.
2.
3.

### Checklist:

- [x] My code follows the code style of this project,
- [ ] My change requires a change to the documentation,
- [x] I described the modification in the CHANGELOG.md file.

### CODE REVIEW

**Review by wojciechczerniak — APPROVED** [review]

File: src/interpreter/plugin/DateTimePlugin.ts [hunk 1]
```diff
@@ -491,6 +542,253 @@ export class DatePlugin extends FunctionPlugin {
...
+    return ans
+  }
+}
+
+/**
+ * Returns i such that:
+ * sortedArray[i-1] < val <= sortedArray[i]
+ *
+ */
+function binsearch(val: number, sortedArray: number[]): number {
```
Comment (bardek8) [inline src/interpreter/plugin/DateTimePlugin.ts:729]: isn't it lowerBound?

File: src/interpreter/plugin/DateTimePlugin.ts [hunk 2]
```diff
@@ -491,6 +542,253 @@ export class DatePlugin extends FunctionPlugin {
...
+
+  private workdaycore(start: number, delta: number, weekend: InternalNoErrorCellValue, holidays?: SimpleRangeValue): InternalScalarValue {
+    start = Math.trunc(start)
+    delta = Math.trunc(delta)
+
+    if(typeof weekend !== 'number' && typeof weekend !== 'string') {
+      return new CellError(ErrorType.VALUE, ErrorMessage.WrongType)
+    }
+
+    let weekendPattern: Maybe<string>
```
Comment (bardek8) [inline src/interpreter/plugin/DateTimePlugin.ts:643]: Wouldn't you prefer to have the `weekend->weekendPattern` mapping as a separate function? If I'm not mistaken we have a copy-pate here.

File: src/interpreter/plugin/DateTimePlugin.ts [hunk 3]
```diff
@@ -491,6 +542,253 @@ export class DatePlugin extends FunctionPlugin {
...
+        return new CellError(ErrorType.NUM, ErrorMessage.BadMode)
+      }
+    }
+
+    const uniqueHolidays = (holidays !== undefined) ? simpleRangeToUniqueNumbers(holidays) : []
+    if(uniqueHolidays instanceof CellError) {
+      return uniqueHolidays
+    }
+
+    const filteredHolidays = uniqueHolidays.filter((arg) => {
```
Comment (bardek8) [inline src/interpreter/plugin/DateTimePlugin.ts:663]: similarly here

**Review by bardek8 — CHANGES_REQUESTED** [review]

File: src/interpreter/plugin/DateTimePlugin.ts [hunk 4]
```diff
@@ -596,76 +596,41 @@ export class DateTimePlugin extends FunctionPlugin {
...
+    return multiplier * this.countWorkdays(start, end, weekendPattern, filteredHolidays)
   }
 
   private workdaycore(start: number, delta: number, weekend: InternalNoErrorCellValue, holidays?: SimpleRangeValue): InternalScalarValue {
     start = Math.trunc(start)
     delta = Math.trunc(delta)
 
-    if(typeof weekend !== 'number' && typeof weekend !== 'string') {
-      return new CellError(ErrorType.VALUE, ErrorMessage.WrongType)
+    if(weekend === '1111111') {
```
Comment (bardek8) [inline src/interpreter/plugin/DateTimePlugin.ts:621]: Do you need it here? IIUC, you'll check it in the next lines anyway.

**Review by bardek8 — APPROVED** [review]
Thanks for the refactor and fixes!

File: src/interpreter/plugin/DateTimePlugin.ts (hunk 4 above)
Comment (izulin) [reply src/interpreter/plugin/DateTimePlugin.ts:621]: thanks for spotting an error. NETWORKDAYS accepts '1111111', while WORKDAY does not

Comment (izulin) [reply src/interpreter/plugin/DateTimePlugin.ts:621]: ok, actually NETWORKDAYS does not accept '1111111' even though it makes sense there...

Comment (bardek8) [reply src/interpreter/plugin/DateTimePlugin.ts]: Yeah, 1111111 is quite an awkward setting :)

**Pull request merged**